### ISSUE TITLE
Bug: Indexing cannot find Obj-C type in mixed module

### ISSUE THREAD

**Issue by thiagohmcruz** [MobileNativeFoundation/rules_xcodeproj#2693]
### Description

In a mixed `rules_ios` module the Obj-C type is causing indexing errors while editing a Swift file. A couple of things I noticed:

* Hit this internally first and for some reason the internal equivalent (using repro branch as a reference) of
```sh
/Users/thiago/Library/Developer/Xcode/DerivedData/rules_ios-aipmacuafufkpzagdnssdupgkczz/Index.noindex/Build/Intermediates.noindex/rules_ios.build/bazel-out/ios-sim_arm64-min15.0-applebin_ios-ios_sim_arm64-dbg-ST-f07cf009171d/bin/iOSApp/Source/iOSApp.library_swift/DerivedSources/swift.compile.params
```
is never created. While repro-ing this in the `rules_xcodeproj` repo that wasn't always the case, only once that file was not created for me, in all other attempts the file was there but the indexing err in the screenshots below was still happening.
* Tried to copy the indexing invocation from Xcode and run it directly on the cmd line but that hasn't given me any errors for some reason. The key missing flag is `-import-underlying-module` and If I manually add that to `OTHER_SWIFT_FLAGS` in Xcode the indexing invocation is 🟢, any chance the indexing invocation in Xcode is not reading the response file `@/.../swift.compile.params` since that's where that flag is located?

### Reproduction steps

* Checkout [this branch](https://github.com/MobileNativeFoundation/rules_xcodeproj/compare/thiago/swift-compile-params-indexing-issue)
* Make sure your DerivedData directory is empty
* `cd examples/rules_ios`
* `bazel run //:xcodeproj` 
* Open `examples/rules_ios/iOSApp/Source/iOSApp.swift`
* Notice the indexing errors similar to the screenshots below

![Screenshot 2023-10-23 at 11 02 13 AM](https://github.com/MobileNativeFoundation/rules_xcodeproj/assets/10197663/3d093e07-cdb7-4502-99ea-f8e5f7ef8171)
![Screenshot 2023-10-23 at 11 02 53 AM](https://github.com/MobileNativeFoundation/rules_xcodeproj/assets/10197663/da9e682f-0c83-4566-a0ab-3321a5e5a18f)

### Expected behavior

No indexing errors.

### rules_xcodeproj version

8d900a9bbfdcceabe779a9fc269d7226bb9121e7

### Xcode version

14.2.0.14C18

### Bazel version

6.4.0rc1

### rules_apple version

what rules_xcodeproj 8d900a9bbfdcceabe779a9fc269d7226bb9121e7 uses

### rules_swift version

what rules_xcodeproj 8d900a9bbfdcceabe779a9fc269d7226bb9121e7 uses

**Comment by thiagohmcruz** [comment]
Found these in system logs. Apparently it's trying to prepend the working directory and this causes it to fail to read the response file so the `-import-underlying-method` flag in that file is not applied?
```sh
[1:getBufferStamp:4887: 0.0041] failed to stat file: /var/tmp/_bazel_thiago/90ec49cd0a280be0e29f918d4d51c649/rules_xcodeproj.noindex/indexbuild_output_base/execroot/register/@/Users/thiago/Library/Developer/Xcode/DerivedData/bazel-project-dulmcsxvjfafxhgioeglsepodhqt/Index.noindex/Build/Intermediates.noindex/bazel-project.build/bazel-out/ios-sim_arm64-min14.1-applebin_ios-ios_sim_arm64-dbg-ST-adb296740065/bin/Frameworks/Foo/Foo_swift/DerivedSources/swift.compile.params (No such file or directory)
```
```sh
[1:fileContentsForFilesInCompilerInvocation:4887: 0.0042] failed getting file contents for /var/tmp/_bazel_thiago/90ec49cd0a280be0e29f918d4d51c649/rules_xcodeproj.noindex/indexbuild_output_base/execroot/register/@/Users/thiago/Library/Developer/Xcode/DerivedData/bazel-project-dulmcsxvjfafxhgioeglsepodhqt/Index.noindex/Build/Intermediates.noindex/bazel-project.build/bazel-out/ios-sim_arm64-min14.1-applebin_ios-ios_sim_arm64-dbg-ST-adb296740065/bin/Frameworks/Foo/Foo_swift/DerivedSources/swift.compile.params: error opening input file '/var/tmp/_bazel_thiago/90ec49cd0a280be0e29f918d4d51c649/rules_xcodeproj.noindex/indexbuild_output_base/execroot/register/@/Users/thiago/Library/Developer/Xcode/DerivedData/bazel-project-dulmcsxvjfafxhgioeglsepodhqt/Index.noindex/Build/Intermediates.noindex/bazel-project.build/bazel-out/ios-sim_arm64-min14.1-applebin_ios-ios_sim_arm64-dbg-ST-adb296740065/bin/Frameworks/Foo/Foo_swift/DerivedSources/swift.compile.params' (No such file or d<…>
```

**Comment by brentleyjones** [comment]
This should be fixed with incremental generation mode (`6b5781b85603141caddfc9ef301a2757a04d68cf` for you to test). I'm not sure what exactly is all needed to make it work with legacy generation, but since we have all Swift compiler flags back in the project, it sidesteps both of those issues.

**Comment by thiagohmcruz** [comment]
FYI confirmed that this issue is not happening in the incremental generator.
